.github/workflows/build-ants-windows.yml

```diff
@@ -0,0 +1,119 @@
+name: Build ANTs
+
+on:
+  pull_request:
+    branches: [ master ]
+  workflow_dispatch:
+    inputs:
+      git_ref:
+        description: 'Upstream ANTsX/ANTs commit ref to check out when building'
+        required: true
+        default: 'b37e8b56e4e518b0f2947f80d0e3701a5ddfe120'
+
+jobs:
+
+  build:
+    strategy:
+      matrix:
+        os: [ windows-2019 ]
+    runs-on: ${{ matrix.os }}
+    defaults:
+      run:
+        shell: bash
+    steps:
+    # first, rewrite OS names to be safe for github artifacts.
+    - name: names
+      run: |
+        OS="${{ matrix.os }}"
+        OS=$(echo "$OS" | sed s/://)
+        echo "ARTIFACT=$OS" >> $GITHUB_ENV
+    - uses: actions/checkout@v3
+      with:
+        repository: ANTsX/ANTs
+        # NB: This commit mirrors the state of the kousu/ANTs repository
+        #     at the time of https://github.com/kousu/ANTs/pull/5.
+        ref: ${{ github.event.inputs.git_ref || 'b37e8b56e4e518b0f2947f80d0e3701a5ddfe120' }}
+    # The step below sets an option inside the cmake file to use the `https` protocol, which addresses:
+    # https://github.blog/2021-09-01-improving-git-protocol-security-github/#no-more-unauthenticated-git
+    - name: Patch in fix for insecure git:// protocol
+      run: |
+        perl -pi -e 's/http instead." ON\)/http instead." OFF\)/g' SuperBuild.cmake
+    - name: cmake generate
+      run: |
+        mkdir antsbin
+        cd antsbin
+        # TODO: check if cmake picks up CMAKE_CXX_FLAGS without being told about it
+        # NB: `BUILD_TESTING=OFF` fixes https://github.com/ANTsX/ANTs/issues/1236
+        # NB: `-g "Unix Makefiles"` is needed to generate files for make (https://stackoverflow.com/a/59551107)
+        cmake \
+         -G "Unix Makefiles" \
+         -DBUILD_TESTING=OFF \
+         -DBUILD_SHARED_LIBS=OFF \
+         -DCMAKE_CXX_FLAGS="$CMAKE_CXX_FLAGS" \
+         -DZLIB_INCLUDE_DIR=D:\lib\zlib \
+         -DZLIB_LIBRARY=D:\lib\zlib\libz.dll.a \
+         ../
+    - name: make
+      working-directory: antsbin
+      run: |
+        ls
+        make VERBOSE=1 -j 4
```

Comment:
During the `make` step, I ran into the exact same issue that I ran into when building ITK locally on Windows:

```
 CMake Error at CMakeLists.txt:109 (message):
  ITK source code directory path length is too long (64 > 50).Please move the
-- Configuring incomplete, errors occurred!
  ITK source code directory to a directory with a shorter path.
See also "D:/a/spinalcordtoolbox-ants/spinalcordtoolbox-ants/antsbin/ITKv5-build/CMakeFiles/CMakeOutput.log".
```

As a short-term fix (as to not impede debugging), I'm going to rename the repository from `spinalcordtoolbox-ants` (22 chars) to `build_ANTs` (10 chars) which save 12x2=24 characters, putting us well under the 50 char limit.

But, in the long-term, it would be nice to see if there are any better workarounds.

Reply:
There may be some hints on [this github issue](https://github.com/SyneRBI/SIRF-SuperBuild/issues/302). In particular, there's a link to the ITK CMakeLists.txt file which includes the 50-character check, so it's an ITK-specific limit. It may be too conservative, since [the Windows limit](https://docs.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=cmd) looks like it's 260 characters (does the ITK hierarchy require 260-50=210 characters?). It looks like the 50-character limit is not configurable, but there *is* an option `ITK_SKIP_PATH_LENGTH_CHECKS` which skips it entirely.

Reply:
!! Thank you for the sleuthing!

Reply:
Thank you from the future, @mguaypaq! Your comment here saved me some trouble while working on https://github.com/jqmcginnis/build_biomedia_stitching/pull/1, as I needed to build ITK once more for that project. :smile: